src/Workspaces/Core/Portable/Workspace/Solution/TextDocumentState.cs

```diff
@@ -77,39 +77,24 @@ private static ITextAndVersionSource CreateStrongText(TextAndVersion text)
         private static ITextAndVersionSource CreateStrongText(TextLoader loader)
             => new LoadableTextAndVersionSource(loader, cacheResult: true);
 
-        private static ITextAndVersionSource CreateRecoverableText(TextAndVersion text, LoadTextOptions loadTextOptions, SolutionServices services)
+        private static ITextAndVersionSource CreateRecoverableText(TextAndVersion text, SolutionServices services)
         {
             var service = services.GetRequiredService<IWorkspaceConfigurationService>();
             var options = service.Options;
 
-            if (options.DisableRecoverableText)
-                return CreateStrongText(text);
-
-            var result = new RecoverableTextAndVersion(new ConstantTextAndVersionSource(text), services);
-
-            if (!options.DeferCreatingRecoverableText)
-            {
-                // This RecoverableTextAndVersion is created directly from a TextAndVersion instance. In its initial state,
-                // the RecoverableTextAndVersion keeps a strong reference to the initial TextAndVersion, and only
-                // transitions to a weak reference backed by temporary storage after the first time GetValue (or
-                // GetValueAsync) is called. Since we know we are creating a RecoverableTextAndVersion for the purpose of
-                // avoiding problematic address space overhead, we call GetValue immediately to force the object to weakly
-                // hold its data from the start.
-                result.GetValue(loadTextOptions, CancellationToken.None);
```

Comment:
the primary reason this AB test existed was because it always felt very unclean to just have this codepath try to force this behavior.  And the aspect of a synchronous method call to get a text-wrapper, forcing synchrounous flushing IO, with a non-cancellable cancellation-token, was always very perturbing.